### ISSUE TITLE
[Bug]: TestChangesHandle3 failed.

### ISSUE THREAD

**Issue by daviszhen** [matrixorigin/matrixone#19650]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

pr commit

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

```
+-----------------------------------------------------------------------------------------------------------------------+
| Failed UT Packages                                                                                                    |
+------------------------------------------------------+----------------------+----------------------+--------+---------+
| PACKAGE                                              | START_TIME           | END_TIME             | RESULT | ELAPSED |
+------------------------------------------------------+----------------------+----------------------+--------+---------+
| github.com/matrixorigin/matrixone/pkg/vm/engine/test | 20[24](https://github.com/matrixorigin/matrixone/actions/runs/11559987952/job/32175884830?pr=19633#step:9:25)-10-28T18:31:46Z | 2024-10-28T18:32:37Z | fail   | 51.48   |
+------------------------------------------------------+----------------------+----------------------+--------+---------+
============================================
[TEST]: TestChangesHandle3
[START_AT]: 2024-10-28T18:31:49Z
[END_AT]: 2024-10-28T18:32:37Z
[RESULT]: fail
[ELAPSED]: 48.58
[OUTPUTS]:
=== RUN   TestChangesHandle3

```

```
2024/10/28 18:32:35.823670 +0000 ERROR jobs/flushTableTail.go:306 [FLUSH-ERR] {"task": "[FT-20]16385-test1", "error": "ExpectedEOB", "phase": "1-merge aobjects"}
2024/10/28 18:32:35.823762 +0000 WARN db/task.go:66 Task[20] exec error: ExpectedEOB
2024/10/28 18:32:36.821228 +0000 INFO cn-service.logtail-server handle subscription asynchronously {"uuid": "", "server-id": "0192d466-06a9-7562-a37f-fadb9a8d2a9d", "table": "db_id:16384 tb_id:16385 "}
panic: runtime error: index out of range [245] with length 13

goroutine 563 [running]:
github.com/matrixorigin/matrixone/pkg/container/types.Date.Calendar(0xd602ed3f, 0x1)
	/home/runner/work/matrixone/matrixone/pkg/container/types/date.go:462 +0x252
github.com/matrixorigin/matrixone/pkg/container/types.Date.String(0xd602ed3f)
	/home/runner/work/matrixone/matrixone/pkg/container/types/date.go:288 +0x37
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/common.TypeStringValue({0x32, 0x0, 0x0, 0x0, 0x4, 0x20, 0x0}, {0x6db1180, 0xc00972769c}, 0x0, ...)
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/common/print.go:145 +0xbcc
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail.ToStringTemplate({0x7cc48d0, 0xc0310ee640}, 0x3, {0xc0124d2940, 0x1, 0x1})
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/tools.go:48 +0x265
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail.BatchToString({0x6fb7b82, 0x4}, 0xc0069e9860, 0x0)
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/tools.go:74 +0x8b8
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail.DebugBatchToString({0x6fb7b82, 0x4}, 0xc0069e9860, 0x0, 0x0)
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/tools.go:57 +0x9d
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail.(*TableLogtailRespBuilder).BuildResp.func1(0x0, 0x0, 0xc0069e9860, 0x0)
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/handle.go:345 +0x5f5
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail.(*TableLogtailRespBuilder).BuildResp(0xc0310ea870)
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/handle.go:396 +0x769
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail.HandleSyncLogTailReq({0x7c05df0, 0xc0310e6cb0}, {0x7faf54e79948, 0xc006ac65a0}, 0xc006aeed00, 0xc0070d4540, {0xc0310ec8a0, 0xc0310ec8d0, 0xc0310e6d20, {}, ...}, ...)
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/handle.go:163 +0x1258
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail.(*LogtailerImpl).TableLogtail(_, {_, _}, {0x4000, 0x4001, 0x0, 0x0, {0x0, 0x0}, {0x0, ...}, ...}, ...)
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/logtailer.go:91 +0x2f0
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail/service.(*LogtailServer).getSubLogtailPhase.func2()
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/service/server.go:517 +0x222
runtime/trace.WithRegion({0x7c05d48, 0xc008546540}, {0x70070c9, 0x19}, 0xc0124d3af8)
	/opt/hostedtoolcache/go/1.23.0/x64/src/runtime/trace/annotation.go:141 +0x138
github.com/matrixorigin/matrixone/pkg/common/moprobe.WithArgAndFn({0x7c05d48, 0xc008546540}, 0x5, 0x0, 0x0, 0xc0124d3af8)
	/home/runner/work/matrixone/matrixone/pkg/common/moprobe/probe.go:99 +0xda
github.com/matrixorigin/matrixone/pkg/common/moprobe.WithRegion(...)
	/home/runner/work/matrixone/matrixone/pkg/common/moprobe/probe.go:88
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail/service.(*LogtailServer).getSubLogtailPhase(0xc008507450, {0x7c05d48, 0xc008546540}, {0xdf8475800, {0xc0310d9095, 0xb}, 0xc0310ec810, 0xc008c8e300}, {0x0, 0x0, ...}, ...)
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/service/server.go:516 +0x45e
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail/service.(*LogtailServer).pullLogtailsPhase1(0xc008507450, {0x7c05d48, 0xc008546540}, {0xdf8475800, {0xc0310d9095, 0xb}, 0xc0310ec810, 0xc008c8e300})
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/service/server.go:418 +0x46b
created by github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logtail/service.(*LogtailServer).logtailPullWorker in goroutine 494
	/home/runner/work/matrixone/matrixone/pkg/vm/engine/tae/logtail/service/server.go:401 +0x6f

```

### Steps to Reproduce

```Markdown
https://github.com/matrixorigin/matrixone/actions/runs/11559987952/job/32175884830?pr=19633
```

**Comment by jiangxinmeng1** [comment]
还没进展

**Comment by jiangxinmeng1** [comment]
fixed by #19763
